src/Storages/StorageS3Settings.cpp

```diff
@@ -26,6 +26,7 @@ void StorageS3Settings::loadFromConfig(const String & config_elem, const Poco::U
         auto endpoint = config.getString(config_elem + "." + key + ".endpoint");
         auto access_key_id = config.getString(config_elem + "." + key + ".access_key_id", "");
         auto secret_access_key = config.getString(config_elem + "." + key + ".secret_access_key", "");
+        auto use_environment_credentials = config.getBool(config_elem + "." + key + ".use_environment_credentials", false);
```

Comment:
oh, it's per bucket? we need also a global setting for the whole server. It can have dozens of buckets, it's silly to ask the user to create a config for each one (and the idea behind that role-based is to be able to manage accesses globally in aws console w/o adjusting config for every bucket).

Reply:
+